daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -760,10 +769,20 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite) {
     case "false" => false
     case _ => false
   }
-  lazy val validationMode: ValidationMode.Type = (testCaseXML \ "@validation").text match {
-    case "" => defaultValidationMode
-    case mode => ValidationMode.fromString(mode)
-  }
+  lazy val validatorName: String = (testCaseXML \ "@validation").text match {
+    case "" => defaultValidatorName
+    case "on" => "xerces"
+    case mode => mode
+  }
+  lazy val validator: api.validation.Validator =
+    Validators
+      .getInstance()
+      .get(validatorName)
+      .make({
+        val props = new Properties()
+        props.setProperty(Validator.rootSchemaKey, getSuppliedSchema().uriForLoading.toString)
+        props
+      })
```

Comment:
Seeing this outside of the CLI makes me reconsider if it would be useful to have a function to auto create a Properties with the rootSchemaKey set. This could all be simplified to a single line. Maybe just document that it's a helper utility, and it's not mandatory, but it does set the rootSchemaKey, which many validators can use.